pallets/democracy-mbm/src/lib.rs

```diff
@@ -339,14 +344,29 @@ impl<T: pallet_democracy::Config + frame_system::Config, W: weights::WeightInfo>
     }
 }
 
+pub struct DemocracyMigrationSaveMigrationBlock<T>(PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for DemocracyMigrationSaveMigrationBlock<T> {
+    fn on_runtime_upgrade() -> Weight {
+        let block_number = frame_system::Pallet::<T>::block_number().saturated_into::<u32>();
+        MigrationBlockNumber::<T>::set(block_number);
+        T::DbWeight::get().reads_writes(1, 1)
+    }
+}
+
 #[frame_support::pallet]
 pub mod pallet {
     use super::*;
+    use frame_support::pallet_prelude::{StorageValue, ValueQuery};
 
     #[pallet::pallet]
     #[pallet::without_storage_info]
     pub struct Pallet<T>(_);
 
     #[pallet::config]
     pub trait Config: frame_system::Config + pallet_democracy::Config {}
+
+    #[pallet::storage]
+    #[pallet::whitelist_storage]
```

Comment:
Not sure this should be whitelisted.

If you decide to keep it so, please account manually for the read weight in the migration logic step (when you read it the first time).

Reply:
Yes better remove it